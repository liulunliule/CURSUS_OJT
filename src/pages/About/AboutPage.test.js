import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
// import '@testing-library/jest-dom/extend-expect';
import AboutPage from './index';

describe('AboutPage', () => {
  test('renders AboutPage correctly', () => {
    render(<AboutPage />);

    expect(screen.getByRole('heading', { name: /Improving Lives Through Learning/i })).toBeInTheDocument();

    ['About', 'Blog', 'Company', 'Careers', 'Press'].forEach(item => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });

    expect(screen.getByText(/Our Features/i)).toBeInTheDocument();
    expect(screen.getByText(/Mobile learning/i)).toBeInTheDocument();
    expect(screen.getByText(/Academic & Technical Support/i)).toBeInTheDocument();
    expect(screen.getByText(/Sharable Certificates/i)).toBeInTheDocument();
    expect(screen.getByText(/An Inclusive Experience/i)).toBeInTheDocument();

    expect(screen.getByText(/Our Story/i)).toBeInTheDocument();

    expect(screen.getByText(/Our Global Reach/i)).toBeInTheDocument();

    expect(screen.getByText(/Meet Our Team/i)).toBeInTheDocument();
  });

  test('navigates to correct page when menu item is clicked', () => {
    render(<AboutPage />);

    delete window.location;
    window.location = { href: '' };

    const menuItems = ['About', 'Company', 'Blog', 'Careers', 'Press'];
    menuItems.forEach(item => {
      fireEvent.click(screen.getByText(item));
      expect(window.location.href).toContain(`/secondLayout/${item.toLowerCase()}`);
    });
  });
});