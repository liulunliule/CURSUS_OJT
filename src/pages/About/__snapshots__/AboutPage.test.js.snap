// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage component renders About component correctly 1`] = `
<DocumentFragment>
  <div
    class="about-page"
  >
    <header
      class="custom-header"
      style="background-color: rgb(248, 248, 248);"
    >
      <nav
        class="custom-nav"
      >
        <ul
          class="custom-ul"
        >
          <li
            class="custom-li selected"
          >
            About
          </li>
          <li
            class="custom-li "
          >
            Blog
          </li>
          <li
            class="custom-li "
          >
            Company
          </li>
          <li
            class="custom-li "
          >
            Careers
          </li>
          <li
            class="custom-li "
          >
            Press
          </li>
        </ul>
      </nav>
    </header>
    <section
      class="hero"
      style="background-color: rgb(248, 248, 248);"
    >
      <h1>
        Improving Lives Through Learning
      </h1>
    </section>
    <section
      class="features"
    >
      <h2>
        Our Features
      </h2>
      <p>
        On Cursus, you have access to:
      </p>
      <div
        class="feature-list"
      >
        <div
          class="feature-item"
        >
          <i
            class="icon mobile-learning"
          />
          <img
            alt="Mobile"
            src="Mobile learning.png"
          />
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold; margin-top: 30px;"
          >
            <h3>
              Mobile learning
            </h3>
          </div>
          <p>
            Quisque nec volutpat sem. Interdum et malesuada fames ac ante ipsum primis in faucibus.
          </p>
        </div>
        <div
          class="feature-item"
        >
          <i
            class="icon support"
          />
          <img
            alt="Academic"
            src="Academic Technical Support.png"
          />
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold; margin-top: 30px;"
          >
            <h3>
              Academic & Technical Support
            </h3>
          </div>
          <p>
            Quisque nec volutpat sem. Interdum et malesuada fames ac ante ipsum primis in faucibus.
          </p>
        </div>
        <div
          class="feature-item"
        >
          <i
            class="icon certificates"
          />
          <img
            alt="Sharable"
            src="Sharable Certificates.png"
          />
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold; margin-top: 30px;"
          >
            <h3>
              Sharable Certificates
            </h3>
          </div>
          <p>
            Quisque nec volutpat sem. Interdum et malesuada fames ac ante ipsum primis in faucibus.
          </p>
        </div>
        <div
          class="feature-item"
        >
          <i
            class="icon inclusive"
          />
          <img
            alt="Inclusive"
            src="An Inclusive Experience.png"
          />
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold; margin-top: 30px;"
          >
            <h3>
              An Inclusive Experience
            </h3>
          </div>
          <p>
            Quisque nec volutpat sem. Interdum et malesuada fames ac ante ipsum primis in faucibus.
          </p>
        </div>
      </div>
    </section>
    <section
      class="our-story"
    >
      <div
        class="our-story-content"
      >
        <div
          class="our-story-text"
        >
          <h2>
            Our Story
          </h2>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed consectetur vel dolor id ultrices. Proin a magna at mi pretium pulvinar in eu enim. Nulla vel lacus lectus. Donec at venenatis augue. Nam vitae purus placerat, hendrerit nisl id, finibus magna. Etiam pharetra gravida ornare. Donec sagittis, ipsum in egestas egestas, dui lorem sollicitudin justo, ut ullamcorper velit neque eu velit. Ut et fringilla elit. Mauris augue augue, auctor a blandit ac, convallis eget neque. Curabitur in ante ante. Nullam laoreet tempus erat at ornare. In nisl nisi, dapibus eget facilisis sit amet, commodo quis nibh.
          </p>
        </div>
        <div
          class="our-story-image"
        >
          <img
            alt="Story"
            src="story.png"
          />
        </div>
      </div>
    </section>
    <section
      class="global-reach"
    >
      <h2>
        Our Global Reach
      </h2>
      <p>
        Cursus is the leading global marketplace for teaching and learning, connecting millions of students to the skills they need to succeed.
      </p>
      <div
        class="reach-stats"
      >
        <div>
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold;"
          >
            <h3>
              25k
            </h3>
          </div>
          <p>
            Instructors
          </p>
        </div>
        <div>
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold;"
          >
            <h3>
              95k
            </h3>
          </div>
          <p>
            Courses
          </p>
        </div>
        <div>
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold;"
          >
            <h3>
              40M
            </h3>
          </div>
          <p>
            Course enrollments
          </p>
        </div>
        <div>
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold;"
          >
            <h3>
              50+
            </h3>
          </div>
          <p>
            Languages
          </p>
        </div>
        <div>
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold;"
          >
            <h3>
              595+
            </h3>
          </div>
          <p>
            Membership Partners
          </p>
        </div>
        <div>
          <div
            style="display: flex; flex-direction: column; align-items: center; font-weight: bold;"
          >
            <h3>
              295
            </h3>
          </div>
          <p>
            Countries
          </p>
        </div>
      </div>
    </section>
    <section
      class="team"
    >
      <h2>
        Meet Our Team
      </h2>
      <p>
        A perfect blend of creativity and technical wizardry. The best people formula for great websites!
      </p>
      <img
        alt="Our Team"
        src="team-photo.jpg"
      />
      <p
        class="team-description"
      >
        Morbi eget elit eget turpis varius mollis eget vel massa. Donec porttitor, sapien eget commodo vulputate, erat felis aliquam dolor, non condimentum libero dolor vel ipsum. Sed porttitor nisi eget nulla ullamcorper eleifend. Fusce tristique sapien nisi, vel feugiat neque luctus sit amet. Quisque consequat quis turpis in mattis. Maecenas eget mollis nisl. Cras porta dapibus est, quis malesuada ex iaculis at. Vestibulum egestas tortor in urna tempor, in fermentum lectus bibendum. In leo leo, bibendum at pharetra at, tincidunt in nulla. In vel malesuada nulla, sed tincidunt neque. Phasellus at massa vel sem aliquet sodales non in magna. Ut tempus ipsum sagittis neque cursus euismod. Vivamus luctus elementum tortor, ac aliquet dolor vehicula et.
      </p>
      <button
        class="join-team"
      >
        Join Our Team
      </button>
    </section>
  </div>
</DocumentFragment>
`;